graphql/src/schema/queries/invoice_counts.rs

```
@@ -31,16 +28,10 @@ fn do_invoice_count(
             timezone_offset,
         )
         .map_err(|err| match err {
-            InvoiceCountError::RepositoryError(_) => Error {
-                message: "InternalError".to_string(),
-                source: None,
-                extensions: None,
-            },
-            InvoiceCountError::BadTimezoneOffset => Error {
-                message: "BadUserInput".to_string(),
-                source: None,
-                extensions: None,
-            },
+            InvoiceCountError::RepositoryError(err) => StandardGraphqlError::from(err),
+            InvoiceCountError::BadTimezoneOffset => {
+                StandardGraphqlError::BadUserInput("Invalid timezone offset".to_string())
```

Hmm, i thought we were going with debug output of service error ? (since we wanted to simplify mapping between service error and api errors)

---

Also it would mean we can avoid indirection of 'from'

---

I see I think I got you slightly wrong. You mean squashing all errors into a single standard error, right? thought we report at least some error code... Guess we can merge internal and bad user input but I think we need a special code for not_authenticated to give the frontend a change to re-login. However not authenticated (and not permitted) only needs to be handled once when validating the request so this wouldn't cause any extra work.

---

I just meant to avoid string literals, as per code snippet here: https://github.com/openmsupply/remote-server/pull/646#discussion_r764809031

---

So just format!("{}", err)? can do, but just wanted to provided a nicer error msg

---

sure, up to you i guess, i like the simple format since it requires least amount of mapping.